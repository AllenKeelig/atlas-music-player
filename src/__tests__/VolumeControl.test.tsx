// src/__tests__/VolumeControl.test.tsx
import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import VolumeControl from "../components/VolumeControls";

test("VolumeControl renders with default volume", () => {
  const { container } = render(
    <VolumeControl volume={0.5} setVolume={() => {}} />
  );
  expect(container).toMatchSnapshot();
});

test("VolumeControl renders muted", () => {
  const { container } = render(
    <VolumeControl volume={0} setVolume={() => {}} />
  );
  expect(container).toMatchSnapshot();
});

test("VolumeControl renders max volume", () => {
  const { container } = render(
    <VolumeControl volume={1} setVolume={() => {}} />
  );
  expect(container).toMatchSnapshot();
});
