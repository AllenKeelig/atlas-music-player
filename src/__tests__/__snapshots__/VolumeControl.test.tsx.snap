// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControl renders max volume 1`] = `
<div>
  <div
    class="flex items-center gap-2 mt-4"
  >
    <span
      class="text-sm text-gray-600"
    >
      ğŸ”ˆ
    </span>
    <input
      class="w-40 accent-black"
      max="100"
      min="0"
      type="range"
      value="100"
    />
    <span
      class="text-sm text-gray-600"
    >
      ğŸ”Š
    </span>
  </div>
</div>
`;

exports[`VolumeControl renders muted 1`] = `
<div>
  <div
    class="flex items-center gap-2 mt-4"
  >
    <span
      class="text-sm text-gray-600"
    >
      ğŸ”ˆ
    </span>
    <input
      class="w-40 accent-black"
      max="100"
      min="0"
      type="range"
      value="0"
    />
    <span
      class="text-sm text-gray-600"
    >
      ğŸ”Š
    </span>
  </div>
</div>
`;

exports[`VolumeControl renders with default volume 1`] = `
<div>
  <div
    class="flex items-center gap-2 mt-4"
  >
    <span
      class="text-sm text-gray-600"
    >
      ğŸ”ˆ
    </span>
    <input
      class="w-40 accent-black"
      max="100"
      min="0"
      type="range"
      value="50"
    />
    <span
      class="text-sm text-gray-600"
    >
      ğŸ”Š
    </span>
  </div>
</div>
`;
